pkg/api/validation/validation.go

```diff
@@ -194,9 +214,8 @@ func ValidateObjectMeta(meta *api.ObjectMeta, requiresNamespace bool, nameFn Val
 	}
 	// if the generated name validates, but the calculated value does not, it's a problem with generation, and we
 	// report it here. This may confuse users, but indicates a programming bug and still must be validated.
-	if len(meta.Name) == 0 {
-		allErrs = append(allErrs, errs.NewFieldRequired("name"))
-	} else {
+	allErrs = append(allErrs, ValidateMinimalName(meta.Name, "name")...)
+	if nameFn != nil && (len(meta.Name) > 0) {
```

Comment:
checking for length here is redundant with your minimal name.  Why not just inline the validate minimal function and make this simpler?